packages/frontend/src/components/LightdashVisualization/VisualizationProvider.tsx

```diff
@@ -124,13 +125,17 @@ const VisualizationProvider: FC<Props> = ({
         );
     }, [explore, resultsData?.metricQuery.dimensions]);
 
+    const dimensionIds = useMemo(() => dimensions.map(fieldId), [dimensions]);
```

Comment:
how come so much is using `useMemo` this won't be that slow

Reply:
it's better to have variables as memos inside a context to avoid rerenderings of components that are subscribed to context. besides, this particular operation does not cost much